linkerd2-proxy/src/rt.rs

```diff
@@ -2,10 +2,18 @@ use tokio::runtime::{Builder, Runtime};
 
 #[cfg(feature = "multicore")]
 pub(crate) fn build() -> Runtime {
-    // The proxy creates an additional admin thread, but it would be wasteful to allocate a whole
-    // core to it; so we let the main runtime consume all cores the process has. The basic scheduler
-    // is used when the threaded scheduler would provide no benefit.
-    match num_cpus::get() {
+    // The proxy creates an additional admin thread, but it would be wasteful to
+    // allocate a whole core to it; so we let the main runtime consume all
+    // available cores. The number of available cores is determined by checking
+    // the environment or by inspecting the host or cgroups.
+    //
+    // The basic scheduler is used when the threaded scheduler would provide no
+    // benefit.
+    let cpus = std::env::var("LINKERD2_PROXY_CORES")
+        .ok()
+        .and_then(|v| v.parse::<usize>().ok())
+        .unwrap_or_else(|| num_cpus::get());
+    match cpus {
```

Comment:
should we maybe use 
```suggestion
    match std::cmp::max(cpus, num_cpus::get()) {
```
or something? so that we don't spawn 2000 threads on the off chance that someone tries to set this like it's a k8s CPU request.

this may not matter _that_ much if we only expect the env var to be used for benchmarking etc rather than being UI, but it does seem like it might be good to not silently screw up benchmark results?